htan_registry/tiffinfo.py

```diff
@@ -0,0 +1,96 @@
+import logging
+import os
+import subprocess
+import re
+
+from synapsegenie.example_filetype_format import FileTypeFormat
+from synapsegenie import process_functions
+
+logger = logging.getLogger(__name__)
+
+here = os.path.abspath(os.path.dirname(__file__))
+
+
+class TiffInfo(FileTypeFormat):
+
+    _filetype = "tiffinfo"
+
+    _process_kwargs = ["databaseSynId"]
+
+    def _validate_filetype(self, filePath):
+        assert os.path.basename(filePath[0]).endswith(".tif{1,2}|.svs")
+
+    def _process(self, df):
+        df.columns = [df.upper() for col in df.columns]
+        return df
+
+    def read_file(self, filePathList):
+        '''
+        Each file is to be read in for validation and processing.
+        This is not to be changed in any functions.
+
+        Args:
+            filePathList:  A list of file paths (Max is 2 for the two
+                           clinical files)
+
+        Returns:
+            df: Pandas dataframe of file
+        '''
+        # This file isn't a dataframe, so just return the filepath
+        filepath = filePathList[0]
+        return filepath
+
+    def process_steps(self, df, newPath, databaseSynId):
+        df = self._process(df)
+        process_functions.updateData(self.syn, databaseSynId, df, self.center,
+                                     toDelete=True)
+        df.to_csv(newPath, sep="\t", index=False)
+        return newPath
+
+    def _validate(self, path_or_df):
+        total_error = []
+        warning = []
+
+        # Validate is bioformats compatiable with showinf
+        showinf_cmd = os.path.join(here, '../bftools/showinf')
+        showinf_exc = subprocess.run([showinf_cmd, path_or_df, "-nopix"], capture_output=True, text=True)
+        # If error, set the set a warning
+        if showinf_exc.returncode:
+            total_error.append(f"showinf failed: {showinf_exc.stdout}")
+
+        # Validate the OME-XML with xmlvalid
+        xmlvalid_cmd = os.path.join(here, '../bftools/xmlvalid')
+        xmlvalid_exc = subprocess.run([xmlvalid_cmd, path_or_df], capture_output=True, text=True)
+        # If error, set the set a warning
+        if xmlvalid_exc.returncode:
+            warning.append("xmlvalid failed: {xmlvalid_exc.stdout}")
+
+        # Get all information on the file from tiffinfo
+        tiffinfo_exc = subprocess.run(["tiffinfo", path_or_df], capture_output=True, text=True)
+        # If error, set the error
+        if tiffinfo_exc.returncode:
+            total_error.append("tiffinfo failed: {tiffinfo_exc.stdout}")
+        # If dates found, make an error
+        finddate = len(re.findall(r'date', tiffinfo_exc.stdout, re.IGNORECASE))
+        # If likely redacted dates are found make a warning instead
+        redacted_dates = len(re.findall(r'1970:01:01 00:00:00', tiffinfo_exc.stdout))
+        if redacted_dates:
+            warning.append(f'{finddate} occurences of DATE in TIFF headers that are potentially redacted to "1970:01:01 00:00:00"')
```

Comment:
Warnings currently aren't as well returned.  I don't think I return an email for valid uploads just yet, so an email doesn't get sent out about errors.  I'm not 100% certain, but warnings also currently aren't stored if the file is valid.